README.md

```diff
@@ -50,17 +52,21 @@ Several template variables are made available to the script running inside the t
 * `title` subject (%s)
 * `messageLines` array of body lines (%b)
 
-`dateFnsFormat` is the date-fns [format](https://date-fns.org/docs/format) function. See the [html-bootstrap](https://github.com/ariatemplates/git-release-notes/blob/master/templates/html-bootstrap.ejs) for sample usage.
+`dateFnsFormat` is the date-fns [format](https://date-fns.org/docs/format) function. See the [html-bootstrap](https://github.com/ariatemplates/git-release-notes/blob/master/templates/html-bootstrap.ejs) for usage example.
+
+`options` the object documented below. Useful for parsing the repository name. See the [html-bootstrap](https://github.com/ariatemplates/git-release-notes/blob/master/templates/html-bootstrap.ejs) for sample usage.
+
+`request` is an instance of [sync-request](https://www.npmjs.com/package/sync-request). This can be useful for querying the Jira API for example to extract extra metadata about a ticket related to a commit.
```

Comment:
I understand the intent but IMHO this would be over complicating `git-release-notes` for something that isn't really so terrible in a custom template.

If we require the template authors to keep their template pure in functional programming sense, then to support that `git-release-notes` has to support these possible scenario's

* (AS-IS) `[git range] -> [git-release-notes] + [git] -> [template] -> [output]`
* (1) `[git range] -> [git-release-notes] + [git] -> [JSON]`
* (3) `[stdin json] -> [git-release-notes] -> [template] -> [output]`

Scenario (3) reduces git-release-notes to just being a template renderer.
Seems like a lot of work to prevent template authors from adding logic to their templates when the template code doesn't look too bad

```
   <div class="row">
        <h2>Jira Tickets</h2>
        <% 
            var jiraTickets = [];
            commits.forEach(function (commit){
                if (commit.jira)
                {
                    jiraTickets.push(commit.jira);
                }
            });
            
            uniqueJira = jiraTickets.filter(function(elem, pos) {
                return jiraTickets.indexOf(elem) == pos;
            });               
        %>

        <table id="jiraTable" class="table table-bordered table-striped">
         <thead>
          <tr>
            <th>Ticket</th>
            <th>Description</th>
            <th>Type</th>
            <th>Status</th>
          </tr>
          </thead>
          <tbody>
            <% 
                process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'; // https://github.com/ForbesLindesay/sync-request/issues/24
                uniqueJira.forEach(function (jira) {
                    var jiraUrl = 'https://atlassian.au.mycorp.net/jira/rest/api/2/issue/' + jira + '?fields=summary,issuetype,status';
                    var jiraApiRequest = request('GET', jiraUrl);
                    var jiraText = jiraApiRequest.getBody();
                    var jiraMetadata = JSON.parse(jiraText);                        
            %>
            <tr>
               <td><a target="jira" href="https://atlassian.au.challenger.net/jira/browse/<%= jira %>"><%= jira %></a></td>
               <td> <%= jiraMetadata.fields.summary %> </td>
               <td> <%= jiraMetadata.fields.issuetype.name %> </td>
               <td> <%= jiraMetadata.fields.status.name %> </td>
            </tr>
            <% }) %>
            </tbody>
          </table>                    
    </div>
```

The `commit.jira` data came from the nice `-t` title parsing feature

That's my 2c :)